src/writer.rs

```diff
@@ -24,16 +24,18 @@ impl<'a> Writer<'a> {
         let request = self.post_request::<AsyncBody>(data)?;
         let mut response = request.send_async().await?;
         let status = response.status();
+        let headers = response.headers().to_owned();
```

Comment:
Since only one header is relevant, maybe extract it here instead of cloning the whole headers set?